cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -520,6 +520,7 @@ impl CoreEnvironment<ReadOnly> {
         flox: &Flox,
         groups_or_iids: &[String],
     ) -> Result<UpgradeResult, CoreEnvironmentError> {
+        tracing::debug!(to_upgrade = groups_or_iids.join(","), "upgrading");
```

Comment:
These debugging statements were helpful when I was figuring out what requests/responses were present in the `/resolve` code path.